.github/workflows/build_and_test.yml

```diff
@@ -0,0 +1,107 @@
+name: build_and_test
+
+on: 
+  push:
+    branches:
+      - main
+  pull_request:
+
+concurrency:
+  group: build_and_test-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+env:
+  RESULT_PATH: testresults
+  PYTHON_VERSION: 3.8.10
+  REQUIREMENTS_PATH: "tests/requirements.txt"
+
+jobs:
+  build:
+    name: build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out the codebase.
+        uses: actions/checkout@v2
+
+      - name: Build image and bundle
+        env:
+          PULL_CACHE: yes
+          AGENT_VERSION: latest
+        run: |
+          make bundle
+          mkdir -p ./dist
+          docker save -o ./dist/image.tar quay.io/signalfx/signalfx-agent-dev:latest
+          mv signalfx-agent-latest.tar.gz ./dist/
+
+      - name: Uploading image artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: image
+          path: ./dist/image.tar
+
+      - name: Uploading bundle artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: bundle
+          path: ./dist/signalfx-agent-latest.tar.gz
+
+  bundle_tests:
+    name: bundle_tests
+    runs-on: ubuntu-latest
+    needs: [build]
+    steps:
+      - name: Check out the codebase.
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Verify Changed files
```

Comment:
Let's run the bundle tests for all pushes to `main` and pull requests since it only takes a few minutes.

For the other test jobs that have not yet been migrated to this workflow, maybe we can try https://github.com/dorny/paths-filter instead of the `changes-include-dir` script?

Reply:
As per your suggestion, I have removed `Verify Changed files` step for `bundle_tests` job in GitHub action.

Yes, I can try with [paths-filter](https://github.com/dorny/paths-filter) to check relevant changes in PR or commit.